packaging/debian-sid/snapd.postrm

```diff
@@ -2,6 +2,9 @@
 
 set -e
 
+# note that we run the debhelper scripts first to ensure snapd is stopped
+#DEBHELPER#
```

Comment:
I think this move is wrong. We should stop snapd.service / socket only after stopping all snap services. The ExecStop command is invoked via snap run and risks waiting indefinitely for skewed system key. In addition the application can still use `snapctl` requests.

I think this should move back to the bottom. We should stop all snap services here, stop snapd and then unmount and remove everything.

Perhaps we need a mode where snapd will not install anything new and will just shut down everything "snap set snapd.maintenance=true" or something?

Reply:
debhelper disables and masks all the services that come from the snapd package, so running it first ensures none of those could get accidentally activated. Additionally, at this point there's no snapd or snap binaries. The stop command won't run because those are executed via `/usr/bin/snap --command=stop ..`. The suspicions I had before about stop command messing things up proved wrong.

Reply:
Given that at this point no snapd/snap binaries are there and the serviecs got stopped in the prerm already it does not really matter if it runs early or later. Unless I miss something. But I think I slightly prefer moving to the top to make very clear that there is really no snapd anymore at this point.